dspace-iiif/src/main/java/org/dspace/app/iiif/service/WordHighlightSolrSearch.java

```diff
@@ -167,28 +160,36 @@ private SolrQuery getSolrQuery(String query, String manifestId) {
     private String getAnnotationList(UUID uuid, String json, String query) {
         searchResult.setIdentifier(manifestId + "/search?q="
                 + URLEncoder.encode(query, StandardCharsets.UTF_8));
-        GsonBuilder builder = new GsonBuilder();
-        Gson gson = builder.create();
-        JsonObject body = gson.fromJson(json, JsonObject.class);
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode body = null;
+        try {
+            body = mapper.readTree(json);
+        } catch (JsonProcessingException e) {
+            log.error("Unable to process json response.", e);
+        }
+        // If error occurred or no body, return immediately
         if (body == null) {
-            log.warn("Unable to process json response.");
             return utils.asJson(searchResult.generateResource());
         }
-        // outer ocr highlight element
-        JsonObject highs = body.getAsJsonObject("ocrHighlighting");
-        // highlight entries
-        for (Map.Entry<String, JsonElement> ocrIds: highs.entrySet()) {
-            // ocr_text
-            JsonObject ocrObj = ocrIds.getValue().getAsJsonObject().getAsJsonObject("ocr_text");
-            // snippets array
-            if (ocrObj != null) {
-                for (JsonElement snippetArray : ocrObj.getAsJsonObject().get("snippets").getAsJsonArray()) {
-                    String pageId = getCanvasId(snippetArray.getAsJsonObject().get("pages"));
-                    for (JsonElement highlights : snippetArray.getAsJsonObject().getAsJsonArray("highlights")) {
-                        for (JsonElement highlight : highlights.getAsJsonArray()) {
-                            searchResult.addResource(getAnnotation(highlight, pageId, uuid));
-                        }
-                    }
+
+        // Example structure of Solr response available at
+        // https://github.com/dbmdz/solr-ocrhighlighting/blob/main/docs/query.md
+        // Get the outer ocrHighlighting node
+        JsonNode highs = body.get("ocrHighlighting");
+
+        // Loop through each highlight entry under ocrHighlighting
+        for (final JsonNode highEntry : highs) {
+            // Get the ocr_text node under the entry
+            JsonNode ocrNode = highEntry.get("ocr_text");
+            // Loop through the snippets array under that
+            for (final JsonNode snippet : ocrNode.get("snippets")) {
+                // Get a canvas ID based on snippet's pages
+                String pageId = getCanvasId(snippet.get("pages"));
+                // Loop through array of highlights for each snippet.
+                for (final JsonNode highlight : snippet.get("highlights")) {
+                    // Add annotation associated with each highlight
```

Comment:
There can be multiple word highlights on a page so the the `highlight` JsonNode is an array. This addition worked or me:

```
for (int i = 0; i < highlight.size(); i++) {
        searchResult.addResource(getAnnotation(highlight.get(i), pageId, uuid));
}
```

Reply:
Thanks @mspalti !  I've just pushed up this change into this PR.  If you could give it an approval (assuming it looks right), I'd appreciate it.